tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -377,6 +379,49 @@ def test_chunking_fast(self):
         self.assertEqual(output, [{"text": ANY(str)}])
         self.assertEqual(output[0]["text"][:6], "ZBT ZC")
 
+    @require_torch
+    @slow
+    def test_mctct(self):
+        speech_recognizer = pipeline(
+            model="speechbrain/m-ctc-t-large",
+            chunk_length_s=10.0,
+            return_timestamps="word",
+            framework="pt",
+        )
+
+        ds = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation").sort("id")
+        audio = ds[40]["audio"]["array"]
+
+        # It should work for a single non-batched input
+        output = speech_recognizer(audio)
+        self.assertEqual(
+            nested_simplify(output),
+            {
+                "text": "A man said to the universe, Sir, I exist.",
+                "chunks": [
+                    {"text": "A", "timestamp": (0.3, 0.33)},
+                    {"text": "man", "timestamp": (0.51, 0.72)},
+                    {"text": "said", "timestamp": (0.93, 1.14)},
+                    {"text": "to", "timestamp": (1.2, 1.29)},
+                    {"text": "the", "timestamp": (1.38, 1.47)},
+                    {"text": "universe,", "timestamp": (1.59, 2.4)},
+                    {"text": "Sir,", "timestamp": (2.67, 3.15)},
+                    {"text": "I", "timestamp": (3.33, 3.36)},
+                    {"text": "exist.", "timestamp": (3.6, 4.59)},
+                ],
+            },
+        )
+
+        # And also for multiple batched inputs
+        n_repeats = 2
+        audio_tiled = np.tile(audio, n_repeats)
+        output = speech_recognizer([audio_tiled], batch_size=2)
+        self.assertEqual(
+            # XXX: This is a probable bug
+            output[0]["text"],
+            "A man said to the universe, Sir, I exist, a man said to the universe, Sirxist.",
```

Comment:
This is the bug.
The first chunk goes all the way to the end of the audio, and has the correct string.
It gets chunked out at the last `Sir` because of the striding. This is expected.
The second chunk, gets the correct string `Sir, I exist`. However the striding forces it to cut out `Sir I e` instead of just `Sir`.

This means the stitching is failing. 

I supposed this is all linked to the initial spectrogram, and is **impossible** to fix.